freedomofpress/securedrop-client
# Fix mistaken default button in login dialog

# Description

Fixes https://github.com/freedomofpress/securedrop-client/issues/1338
Follow up on #1331 

# Details

The buttons can be defined as "default button", ONLY when they
are children of a QDialog. That was the missing part.

Note that both buttons are "auto default", that is why it is
necessary to specify which one should take precendence by
calling its setDefault method. Otherwise, the rendering order
of the buttons would determine which one is activated when
the journalist presses the Enter key.

See https://doc.qt.io/qt-5/qdialog.html#default-button
and https://doc.qt.io/qt-5/qpushbutton.html#default-prop

# Test Plan

1. Launch the SecureDrop Client
2. In the login dialog, fill in the credentials
3. Press <kbd>Enter</kbd>
- [ ] Confirm that the login form is submitted

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:


If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:


If these changes modify the database schema, you should include a database migration. Please check as applicable:
